File: Python3/프로그래머스/2/42885. 구명보트/구명보트.py
from collections import deque

def solution(people, limit):
    answer = 0
    first = 0
    last = len(people)-1
    people = deque(sorted(people))
    
    while first <= last:
        if people[first] + people[last] <= limit:
            first = first + 1
            last = last - 1
            answer = answer + 1
        else:
            last = last - 1
            answer = answer + 1
            
    return answer

File: C++/프로그래머스/2/42885. 구명보트/구명보트.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> people, int limit) {
    int answer = 0;
    sort(people.begin(), people.end(), [](int a, int b){ return a < b;});
    int left = 0, right = people.size() - 1;
    
    while (left <= right) {
        if (left == right) return ++answer;
        if (people[left] + people[right] <= limit) left++;
        right--;
        answer++;
    }
    
    return answer;
}